### PR TITLE
ref(api): Feature gate org group index multi project

### PR DESCRIPTION
**Pull request by macqueen**
not even totally sure we need to feature gate the methods other than get -- lmk what y'all think

cc @logicalbomb

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
This looks good, although I'm wondering if it makes sense to just build this logic directly into `get_projects`? I'm not sure if there are any situations where it should return > 1 project if an org doesn't have the `sentry10` feature.

Also just realized that if the organization has temporarily opted out from sentry 10 then they'll be unable to query across multiple projects as well. Not sure if there are complications there?

File: src/sentry/api/endpoints/organization_group_index.py [hunk 1]
```diff
@@ -104,6 +105,12 @@ def get(self, request, organization):
         if not projects:
             return Response([])
 
+        if len(projects) > 1 and not features.has(
+                'organizations:sentry10', organization, actor=request.user):
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_group_index.py]: The frontend checks for the `global-views` feature as that is the billing related feature that toggles whether or not a customer's plan allows them to view data from more than one project at a time.

**Comment by macqueen** [comment]
@wedamija yeah i kind of had that thought, but i think there might be some exceptions. the one i could think of/find was in the releases endpoints: https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization.py#L283-L287

we could add a kwarg to `get_projects` like `enforce_feature_check=True` and then handle within get_projects?

File: src/sentry/api/endpoints/organization_group_index.py (hunk 1 above)
Comment (macqueen) [reply src/sentry/api/endpoints/organization_group_index.py]: updated, ty!

**Comment by wedamija** [comment]
> @wedamija yeah i kind of had that thought, but i think there might be some exceptions. the one i could think of/find was in the releases endpoints: https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization.py#L283-L287
> 
> we could add a kwarg to `get_projects` like `enforce_feature_check=True` and then handle within get_projects?

Hmm, I'm not sure that filtering by the results from `get_projects` is actually the right thing to do there. Since we see `include_all_accessible=True`, we actually just want all projects that the user is able to access. Should we just be using `request.access.projects` instead? 

I think that actually, `include_all_accessible` is probably something we should look to remove, since it's a hack I introduced just to standardize our calls to `get_projects`, but realistically I had no idea how our permissions worked at the time. Since this is the only use of it, we could probably just remove it if we switch to using `request.access.projects`.

**Comment by macqueen** [comment]
@wedamija re: your last comment, I think the issue with relying on `request.access.projects` is that for the releases endpoints, we still need to rely on the part of `get_projects` that handles pulling the project ids out of the query params and filtering by them, right? so we'd have to duplicate that logic somewhere. I guess the issue is that release endpoints are different in that they were "global"/multi-project before sentry 10 -- we just recently added the frontend to match that + filtering by multiple projects.

**Comment by wedamija** [comment]
> @wedamija re: your last comment, I think the issue with relying on `request.access.projects` is that for the releases endpoints, we still need to rely on the part of `get_projects` that handles pulling the project ids out of the query params and filtering by them, right? so we'd have to duplicate that logic somewhere. I guess the issue is that release endpoints are different in that they were "global"/multi-project before sentry 10 -- we just recently added the frontend to match that + filtering by multiple projects.

Dang, yeah you're right about that. Let's just leave as is for now.

**Review by markstory — APPROVED** [review]
Looks good. The flag used here is the same as is used on the frontend.

**Pull request merged**